awsx/utils.ts

```diff
@@ -82,15 +82,12 @@ export function getRegionFromOpts(opts: pulumi.CustomResourceOptions): pulumi.Ou
 
 /** @internal */
 export function getRegion(res: pulumi.Resource): pulumi.Output<aws.Region> {
-  // A little strange, but all we're doing is passing a fake type-token simply to get
-  // the AWS provider from this resource.
-  const provider = res.getProvider ? res.getProvider("aws::") : undefined;
-  return getRegionFromProvider(provider);
+  // uses the provider from the parent resource to fetch the region
+  return aws.getRegionOutput({}, { parent: res }).apply(region => region.name as aws.Region);
 }
 
 function getRegionFromProvider(provider: pulumi.ProviderResource | undefined) {
```

Comment:
Specify return type here please. I wonder what the "default" value was, aws.config.region. Now with this change it will pull the ambient AWS provider if provider=undefined, sounds like it's intended.